app/services/dialog_local_service.rb

```diff
@@ -70,27 +30,34 @@ def determine_dialog_locals_for_custom_button(obj, button_name, resource_action,
       :finish_submit_endpoint => cancel_endpoint,
       :cancel_endpoint        => cancel_endpoint,
       :open_url               => false
-    )
-
-    dialog_locals
+    }
   end
 
   private
 
   def determine_api_endpoints(obj, display_options = {})
-    base_name = obj.class.name.demodulize
+    base_name = obj.class.base_model.name
     case base_name
     when /EmsCluster/
       api_collection_name = "clusters"
       cancel_endpoint = "/ems_cluster"
+    when /MiqTemplate/
+      if obj.class.name.demodulize == "Vm"
+        api_collection_name = "vms"
+        cancel_endpoint = display_options[:cancel_endpoint] || "/vm_infra/explorer"
+      else
+        api_collection_name = "templates"
+        cancel_endpoint = display_options[:cancel_endpoint] || "/vm_or_template/explorer"
+      end
+
     when /GenericObject/
       api_collection_name = "generic_objects"
       cancel_endpoint = if !display_options.empty? && display_options[:display_id]
                           "/service/show/#{display_options[:display_id]}?display=generic_objects"
                         else
                           "/service/explorer"
                         end
-    when /InfraManager/
+    when /ExtManagementSystem/
```

Comment:
Do we need to handle `CloudManager` somewhere like we were handling `InfraManager` before?